image-qemu.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2018 Alexandre Fournier <alexandre.fournier@kiplink.fr>, Kiplink
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <confuse.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "genimage.h"
+
+struct qemu {
+	const char *format;
+};
+
+static int qemu_generate(struct image *image)
+{
+	struct partition *part;
+	struct qemu *qemu = image->handler_priv;
+	char *partitions = NULL;
+	int ret;
+
+	list_for_each_entry(part, &image->partitions, list) {
+		struct image *child;
+		const char *infile;
+
+		image_log(image, 1, "adding partition '%s'%s%s%s ...\n", part->name,
+			part->image ? " from '": "",
+			part->image ? part->image : "",
+			part->image ? "'" : "");
+
+		child = image_get(part->image);
```

Comment:
``part->image`` must not be null here but there is a null check above. This is inconsistent. I think not having an image is invalid for qemu images. Please add a loop to check this in the setup function and abort with an appropriate error message. Then you can drop the checks above.